### PR TITLE
Pass TLS credentials to Ostree binary

### PR DESCRIPTION
**Pull request by taheris**
Additional commits also fix Url joins with trailing slashes and sends an installation report to director.

### CODE REVIEW

File: src/datatype/ostree.rs [hunk 1]
```diff
@@ -81,17 +82,19 @@ impl OstreePackage {
...
-            .env("REF_NAME", self.refName.clone())
-            .env("DESCRIPTION", self.description.clone())
-            .env("PULL_URI", self.pullUri.clone());
-        token.map(|t| command.env("AUTHPLUS_ACCESS_TOKEN", t.access_token.clone()));
+        let mut cmd = Command::new("sota_ostree.sh");
+        cmd.env("COMMIT", self.commit.clone());
+        cmd.env("REF_NAME", self.refName.clone());
+        cmd.env("DESCRIPTION", self.description.clone());
+        cmd.env("PULL_URI", self.pullUri.clone());
+        creds.access_token.map(|t| cmd.env("AUTHPLUS_ACCESS_TOKEN", t.clone()));
```
Comment (heartsucker) [inline src/datatype/ostree.rs]: Just double checking. These env vars are only the path to the file, not the contents, correct?

Comment (taheris) [inline src/datatype/ostree.rs]: yep

File: src/datatype/network.rs [hunk 2]
```diff
@@ -53,7 +53,14 @@ pub struct Url(pub url::Url);
 impl Url {
     /// Append the string suffix to this URL.
     pub fn join(&self, suffix: &str) -> Url {
-        Url(self.0.join(suffix).unwrap_or_else(|err| {
+        let mut url = self.0.clone();
+
+        // add trailing slash if it doesn't exist
+        if url.path_segments().map_or(false, |mut segs| segs.next_back().unwrap() != "") {
```
Comment (jochenschneider) [inline src/datatype/network.rs]: Can this be simplified by putting the push inside of the map?

Comment (taheris) [inline src/datatype/network.rs]: Not easily, no :) The problem is that `path_segments` and `path_segments_mut` return different types, and the mut version has no `peek` method so a pop would be destructive and need to be re-added.

Comment (jochenschneider) [inline src/datatype/network.rs]: That's what I feared. How about `url.path_segments_mut().unwrap().pop_if_empty().extend(&[suffix]);`

Comment (taheris) [inline src/datatype/network.rs]: Nice, yeah that's better.

**Pull request merged**